activerecord/test/cases/tasks/database_tasks_test.rb

```diff
@@ -191,79 +207,128 @@ def test_creates_configurations_with_blank_hosts
         ActiveRecord::Tasks::DatabaseTasks.create_all
       end
     end
+
+    private
+
+      def with_stubbed_configuration_establish_connection
+        if current_adapter?(:SQLite3Adapter) && in_memory_db?
+          ActiveRecord::Base.stub(:configurations, @configurations) do
+            # To refrain from connecting to a newly created empty DB in sqlite3_mem tests
+            ActiveRecord::Base.connection_handler.stub(:establish_connection, nil) do
+              yield
+            end
+          end
+        else
+          ActiveRecord::Base.stub(:configurations, @configurations) do
+            yield
+          end
+        end
+      end
   end
 
   class DatabaseTasksCreateCurrentTest < ActiveRecord::TestCase
     def setup
       @configurations = {
         "development" => { "database" => "dev-db" },
         "test"        => { "database" => "test-db" },
-        "production"  => { "url" => "prod-db-url" }
+        "production"  => { "url"      => "prod-db-url" }
```

Comment:
we don't need to align this. It is better to not do.

Reply:
Fixed.